test/note_test.rb

```diff
@@ -2,19 +2,25 @@
 
 class TestNote < MiniTest::Test
   def setup
-    @frequency   = 220.to_f
-    @duration    = 2.0
-    @sample_rate = 22050
-    @note = MusicTheory::Note.new(frequency: @frequency, duration: @duration)
+    @test_samples = eval(File.read('test/fixtures/note_samples_array'))
+    @basic_note = MusicTheory::Note.new
+    @note_220 = MusicTheory::Note.new(frequency: 220)
+    @distorted_note = MusicTheory::Note.new(distort: true)
   end
 
-  def test_total_frames
-    frames = (@duration * @sample_rate).to_i
-    assert_equal frames, @note.total_frames
+  def test_samples_are_identical
+    assert_equal @test_samples, @basic_note.samples
```

Comment:
Thought about this a little more. Another option would be to compare them sample by sample and build up a sum of the difference between each sample. This would allow you specify a tolerance level for taking into account differences in floating point rounding, which why I think this test fails on my machine.